docs: update README to reflect docker-opam-action rename (#107)
coq-community/docker-coq-action: README.md

Reviewer: ```suggestion
# Docker-Opam GitHub Action
```

Reviewer: ```suggestion
[rocq-community/templates](https://github.com/rocq-community/templates#readme).
```

Reviewer: ```suggestion
[rocq-prover/opam](https://github.com/rocq-prover/opam) repository, and
```

Reviewer: ```suggestion
available in <https://rocq-prover.org/docs/opam-packaging>.
```

[python-ci-badge]: https://github.com/rocq-prover/docker-opam-action/actions/workflows/python-demo.yml/badge.svg?branch=master
[rt-ci-badge]: https://github.com/rocq-prover/docker-opam-action/actions/workflows/gha-rt.yml/badge.svg?branch=master
[reviewdog-link]: https://github.com/rocq-prover/docker-opam-action/actions/workflows/reviewdog.yml
[docker-coq-ci-link]: https://github.com/rocq-prover/docker-opam-action/actions/workflows/coq-demo.yml
[python-ci-link]: https://github.com/rocq-prover/docker-opam-action/actions/workflows/python-demo.yml
[rt-ci-link]: https://github.com/rocq-prover/docker-opam-action/actions/workflows/gha-rt.yml
[coqorg-shield]: https://img.shields.io/badge/depends%20on-coqorg%2Fcoq-blue.svg
[coqorg-link]: https://hub.docker.com/r/coqorg/coq
[mathcomp-shield]: https://img.shields.io/badge/see%20also-mathcomp%2Fmathcomp-blue.svg
[mathcomp-link]: https://hub.docker.com/r/mathcomp/mathcomp
[example-shield]: https://img.shields.io/badge/see%20also-example-brightgreen.svg
[example-link]: https://github.com/erikmd/docker-coq-github-action-demo
[contributing-shield]: https://img.shields.io/badge/contributions-welcome-%23f7931e.svg

> [contributing-link]: https://github.com/coq-community/manifesto/blob/master/CONTRIBUTING.md

Reviewer: This link should be removed or replaced.

[conduct-shield]: https://img.shields.io/badge/%E2%9D%A4-code%20of%20conduct-%23f15a24.svg

> [conduct-link]: https://github.com/coq-community/manifesto/blob/master/CODE_OF_CONDUCT.md

Reviewer: Same. In this case, we can point to https://rocq-prover.org/policies/code-of-conduct.

which in turn are based on [rocq/base](https://hub.docker.com/r/rocq/base/),
a Docker image with a Debian environment.

|     | GitHub repo                                                             | Type          | Docker Hub                                                       |
| --- | ----------------------------------------------------------------------- | ------------- | ---------------------------------------------------------------- |
| ⊙   | [docker-opam-action](https://github.com/rocq-prover/docker-opam-action) | GitHub Action | N/A                                                              |

> | ↳   | [docker-rocq](https://github.com/coq-community/docker-rocq)             | Dockerfile    | [`rocq/rocq-prover`](https://hub.docker.com/r/rocq/rocq-prover/) |

Reviewer: ```suggestion
| ↳   | [docker-rocq](https://github.com/rocq-community/docker-rocq)             | Dockerfile    | [`rocq/rocq-prover`](https://hub.docker.com/r/rocq/rocq-prover/) |
```

> | ↳   | [docker-base](https://github.com/coq-community/docker-base)             | Dockerfile    | [`rocq/base`](https://hub.docker.com/r/rocq/base/)               |

Reviewer: ```suggestion
| ↳   | [docker-base](https://github.com/rocq-community/docker-base)             | Dockerfile    | [`rocq/base`](https://hub.docker.com/r/rocq/base/)               |
```

| ↳   | Debian                                                                  | Linux distro  | [`debian`](https://hub.docker.com/_/debian/)                     |

For more details about these images, see the

> [docker-coq wiki](https://github.com/coq-community/docker-coq/wiki).

Reviewer: ```suggestion
[docker-coq wiki](https://github.com/rocq-community/docker-coq/wiki).
```

## OPAM

The `docker-opam-action` provides built-in support for `opam` builds.

> `coq` is built on-top of `ocaml` and so `coq` projects use `ocaml`'s

Reviewer: ```suggestion
Rocq is built on-top of OCaml and so Rocq projects use OCaml's
```

### Versioning

The Git repo of `docker-opam-action` uses `master` as developing branch
and `v1` as release branch; and the corresponding tags `v1.x.y` follow
[semantic versioning](https://semver.org/).

We develop `docker-opam-action` with a special focus on backward
compatibility, so that if your workflow just uses
**`rocq-prover/docker-opam-action@v1`**, you will be able to benefit
from new features, while expecting no breaking change.

However, we recall that the version of any GitHub Action can just as
well be [referenced by a tag or a commit SHA](https://docs.github.com/en/actions/learn-github-actions/finding-and-customizing-actions#using-release-management-for-your-custom-actions).

Contrary to some custom practice of GitHub Actions maintainers, we do not change to which commit a tag points once it is published.
As a result, the latest stable version denoted by the short Git reference `v1` is implemented as a _release branch_, not as a tag.
Anyway, if you do not trust the maintainers of a given GitHub Action, it is always safer to reference a _commit SHA_.

### Inputs

#### `opam_file`

> _Optional_

Reviewer: BTW, you have a lot of spurious changes in the diff like this one. If you can configure your auto-formatter to use `*` instead of `_` for italic, it would reduce significantly the size of the diff.

**Warning!** not all OCaml versions are available with all Coq versions.

The supported compilers w.r.t. each version of Coq are documented in the

> [OCaml-versions policy](https://github.com/rocq-prover/docker-coq/wiki#ocaml-versions-policy) section of the `docker-coq` wiki.

Reviewer: This link shouldn't have been changed (until this wiki moves elsewhere).
